core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/OnDemandRepairJob.java

```diff
@@ -134,12 +135,25 @@ public RepairConfiguration getRepairConfiguration()
 
     public RepairJobView getView()
     {
-        return new RepairJobView(getId(), myOngoingJob.getTableReference(), myRepairConfiguration, null, getStatus(), getProgress());
+        return new OnDemandRepairJobView(
+                getId(),
+                myOngoingJob.getTableReference(),
+                myRepairConfiguration, getStatus(),
+                getProgress(),
```

Comment:
Would it be feasable to refactor Status + Progress and Completion time into a class that can be passed here? I could see some refactoring that could make the view call a little less cluttered.

Reply:
I'm not sure, I think it just got messier. But there is a missing new line on line 141, fixing that improves things a litttle bit.